pkgs/build-support/docker/default.nix

```diff
@@ -757,6 +759,18 @@ rec {
 
       contentsList = if builtins.isList contents then contents else [ contents ];
 
+      # If a fromImage is provided, we prepend its layers to the resulting
+      # image. This untars that fromImage, if any.
+      baseLayers = runCommand "${baseName}-base-layers" {
+        inherit fromImage;
+      } ''
+        mkdir $out
+        if [[ -n "$fromImage" ]]; then
+          echo "Unpacking base image..."
+          tar -C $out -xpf "$fromImage"
+        fi
+      '';
+
```

Comment:
This creates a store path for an intermediate value. If I read correctly, it is not retained as a reference in the final streamLayeredImage or buildLayeredImage output. Would it be possible to move this logic into streamLayeredImage? That way the intermediate files don't have to wait for gc and they don't need to be compressed and uploaded by `cachix push --watch` and Hercules CI.

Reply:
It does, however I'd argue there may be a couple reasons why that's actually desirable: the base image will likely be hoisted on a 3rd party server, which could go down, delete the base image, or make it otherwise unavailable (e.g. Dockerhub DL limits).  So we probably do want to pull it into our nix store and nix binary cache for robustness.  Moreover, unlike the user application nix containers, which change whenever a nix package changes in the closure, the base image will only change when the user changes it in the nix expression, so it'll cause less churn on disk.

Reply:
That makes more sense, yes.

I moved the `fromImage` untar to `stream_layered_image.py` where I create a temporary folder to extract the archive.